gtsam/slam/dataset.h

```diff
@@ -173,14 +173,20 @@ typedef std::pair<size_t, size_t> SIFT_Index;
 
 /// Define the structure for the 3D points
 struct SfM_Track {
-  SfM_Track():p(0,0,0) {}
+  SfM_Track(): p(0,0,0) {}
   Point3 p; ///< 3D position of the point
   float r, g, b; ///< RGB color of the 3D point
   std::vector<SfM_Measurement> measurements; ///< The 2D image projections (id,(u,v))
   std::vector<SIFT_Index> siftIndices;
   size_t number_measurements() const {
     return measurements.size();
   }
+  SfM_Measurement measurement(size_t idx) const {
```

Comment:
docstrings needed. Also add them for constructors while you're at it. Can be
```
/// Construct empty track
```

I must say I don't understand the point of `point` anymore. I seems that can only work in an imperative context :-(